gateway-js/src/index.ts

```diff
@@ -389,10 +399,9 @@ export class ApolloGateway implements GraphQLService {
       throw e;
     }
 
-    this.schema = schema;
     // TODO(trevor): #580 redundant parse
     this.parsedSupergraphSdl = parse(supergraphSdl);
```

Comment:
Hmmm, looking at the code there's two things that make me a bit wary about moving it into `updateWithSchemaAndNotify()`:
- Based on this block:
    ```typescript
    // TODO(trevor): #580 redundant parse
    // This may throw, so we'll calculate early (specifically before making any updates)
    // In the case that it throws, the gateway will:
    //   * on initial load, throw the error
    //   * on update, log the error and don't update
    const parsedSupergraphSdl = parse(result.supergraphSdl);
    ```
    It looks like we parse early to avoid accidental state update (we could double-parse, but I think we've been trying to avoid redundant parses).
- We don't update `this.parsedSupergraphSdl` in `updateByComposition()`, which initially gave me the impression we only update it when we're handed a core supergraph schema directly and not when we get a service list. But looking at `loadStatic()`, we do actually update `this.parsedSupergraphSdl` when running `this.createSchemaFromServiceList()`, so I'm not sure why we don't update it in `updateByComposition()`. Maybe @trevor-scheer might know more here?